pyclam/manifold.py

```diff
@@ -608,6 +605,48 @@ def __repr__(self) -> str:
     def depth(self) -> int:
         return len(self.graphs) - 1
 
+    def distance(self, *, x1: Union[int, List[int], None], x2: Union[int, List[int], None], p1: Data = None, p2: Data = None) -> np.ndarray:
+        """ Calculates the pairwise distances between all points in x1 and x2.
+
+        This DOES NOT do any batching.
+
+        The metric given to Manifold should have the following properties:
+            * dist(p1, p2) = 0 if and only if p1 = p2.
+            * dist(p1, p2) = dist(p2, p1)
+
+        :param x1: index of point or list of indexes of points in self.data.
+        :param x2: index of point or list of indexes of points in self.data.
+        :param p1: (optional) point(s) given directly to override x1. Useful when the point is not in self.data
```

Comment:
weird, I find this p1 p2 notion unintuitive, I'd say make it work with just x1 x2 and make them positional only